encode/av1encode.c

```diff
@@ -1512,6 +1512,45 @@ render_fr_buffer()
     }
 }
 
+static void
+render_quality_buffer()
+{
+    VABufferID param_buf = VA_INVALID_ID;
+    VAStatus va_status;
+    VABufferID render_id = VA_INVALID_ID;
+
+    VAEncMiscParameterBuffer *misc_param;
+    VAEncMiscParameterBufferQualityLevel *misc_quality;
+
+    va_status = vaCreateBuffer(va_dpy, context_id,
+                            VAEncMiscParameterBufferType,
+                            sizeof(VAEncMiscParameterBuffer) + sizeof(VAEncMiscParameterBufferQualityLevel),
+                            1, NULL, &param_buf);
+    CHECK_VASTATUS(va_status, "vaCreateBuffer");
+
+    vaMapBuffer(va_dpy, param_buf, (void **)&misc_param);
+    misc_param->type = VAEncMiscParameterTypeQualityLevel;
+    misc_quality = (VAEncMiscParameterBufferQualityLevel *)misc_param->data;
+    memset(misc_quality, 0, sizeof(*misc_quality));
+
+    //quality_level range {1...7}, 1 is best quality, 7 is best performance
+    // 7 is default setting
+    misc_quality->quality_level = 7;
```

Comment:
suppose this value should come from user?
if you are adding a setting to tune it